noir-projects/aztec-nr/aztec/src/keys/point_to_symmetric_key.nr

```diff
@@ -0,0 +1,32 @@
+use dep::protocol_types::{constants::GENERATOR_INDEX__SYMMETRIC_KEY, grumpkin_point::GrumpkinPoint, utils::arr_copy_slice};
+use dep::std::{hash::sha256, grumpkin_scalar::GrumpkinScalar, scalar_mul::variable_base_embedded_curve};
+
+// TODO(#5726): This function is called deriveAESSecret in TS. I don't like point_to_symmetric_key name much since
+// point is not the only input of the function. Unify naming with TS once we have a better name.
+pub fn point_to_symmetric_key(secret: GrumpkinScalar, point: GrumpkinPoint) -> [u8; 32] {
+    let shared_secret_fields = variable_base_embedded_curve(point.x, point.y, secret.low, secret.high);
+    // TODO(https://github.com/AztecProtocol/aztec-packages/issues/6061): make the func return Point struct directly
+    let shared_secret = GrumpkinPoint::new(shared_secret_fields[0], shared_secret_fields[1]);
+    let mut shared_secret_bytes_with_separator = [0 as u8; 65];
+    shared_secret_bytes_with_separator = arr_copy_slice(shared_secret.to_be_bytes(), shared_secret_bytes_with_separator, 0);
```

Comment:
Why the 0 at the end, could it not use the `generator_index__symmetric_key` directly?

Reply:
My brain was thinking 
```
[shared_secret.to_be_bytes(), shared_secret_bytes_with_separator, 0]
```